frontend/src/views/MockExams/MockExamsList/MockExamsList.tsx

```diff
@@ -4,42 +4,91 @@ import {
 import useSWR from 'swr';
-import { FullPageLoading } from '../../../components/FullPageLoading/FullPageLoading';
+import { useMemo, useState } from 'react';
 import { GeneralAPIError } from '../../../components/GeneralAPIError/GeneralAPIError';
-import { formatLong } from '../../../utils/date';
+import { formatDatesRange, formatLong } from '../../../utils/date';
 import { PERFECT_SCORE } from '../MockExams.constants';
+import { usePagination } from '../../../hooks/usePagination/usePagination';
+import { useSort } from '../../../hooks/useSort/useSort';
+import { addQueryParams } from '../../../utils/addQueryParams';
+import { TableFilters } from '../../../components/Table/TableFilters';
+import { Table } from '../../../components/Table/Table';
+import { DateFilter } from '../../../components/Table/Filters/DateFilter/DateFilter';
+import { IntegerRangeFilter } from '../../../components/Table/Filters/IntegerRangeFilter/IntegerRangeFilter';
+import { BooleanFilter } from '../../../components/Table/Filters/BooleanFilter/BooleanFilter';
 
+const MIN_SCORE = 0;
 export const MockExamsList = () => {
   const pageTitle = 'Próbne egzaminy teoretyczne';
+  const { rowsPerPage, currentPage, onPageChange, onRowsPerPageChange } =
+    usePagination();
+  const {
+    sortOrder,
+    sortedBy,
+    getCellSortDirection,
+    getLabelIsActive,
+    getLabelSortDirection,
+    onSortClick,
+  } = useSort<Required<MockExamAttemptFindAllQueryDto>['sortBy']>(
+    'attemptDate',
+    'desc',
+  );
+  const { isPassed, score, date } = useFilters();
+  const [openedFilter, setOpenedFilter] = useState<null | string>(null);
 
+  const apiUrl = useMemo(
+    () =>
+      addQueryParams<MockExamAttemptFindAllQueryDto>('/api/exams', {
+        page: currentPage,
+        pageSize: rowsPerPage,
+        sortOrder,
+        sortBy: sortedBy,
+        filters: {
+          dateFrom:
+            date.value.from !== undefined
+              ? formatISO(date.value.from)
+              : undefined,
+          dateTo:
+            date.value.to !== undefined
+              ? formatISO(endOfDay(date.value.to))
+              : undefined,
+          isPassed: isPassed.value,
+          scoreFrom: score.value.from,
+          scoreTo: score.value.to,
+        },
+      }),
+    [
+      currentPage,
+      rowsPerPage,
+      sortOrder,
+      sortedBy,
+      date.value,
+      score.value,
+      isPassed.value,
+    ],
+  );
   const { data: mockExamsListData, error: mockExamsListError } =
-    useSWR<MockExamAttemptFindAllResponseDto>('/api/exams');
+    useSWR<MockExamAttemptFindAllResponseDto>(apiUrl);
 
   const navigate = useNavigate();
 
   if (mockExamsListError) {
     return <GeneralAPIError />;
   }
 
-  if (mockExamsListData === undefined) {
-    return <FullPageLoading />;
-  }
-
-  const { examAttempts } = mockExamsListData;
+  const examAttempts = mockExamsListData?.examAttempts;
+  const totalRows = mockExamsListData?.total ?? 0;
```

Comment:
Let's wrap the total in the `useListTotal` hook.
All you have to do is change this into
```js
  const totalRows = useListTotal(instructorsData?.total)
```
and put it before any returns